mne_bids/utils.py

```diff
@@ -621,3 +623,74 @@ def _find_matching_sidecar(bids_fname, bids_root, suffix, allow_fail=False):
         return None
     else:
         raise RuntimeError(msg)
+
+
+def _update_sidecar(sidecar_fname, key, val):
+    """Update a sidecar JSON file with a given key/value pair.
+
+    Parameters
+    ----------
+    sidecar_fname : str
+        Full name of the data file
+    key : str
+        The key in the sidecar JSON file. E.g. "PowerLineFrequency"
+    val : str
+        The corresponding value to change to in the sidecar JSON file.
+    """
+    with open(sidecar_fname, "r") as fin:
+        sidecar_json = json.load(fin)
+    sidecar_json[key] = val
+    with open(sidecar_fname, "w") as fout:
+        json.dump(sidecar_json, fout)
+
+
+def _estimate_line_noise(raw):
+    """Estimate power lin noise from a given BaseRaw.
+
+    Parameters
+    ----------
+    raw : mne.io.BaseRaw
+
+    Returns
+    -------
+    powerlinefrequency : int
+        Either 50, or 60 Hz depending if European,
+        or USA data recording.
+    """
+    temp_raw = raw.copy()
+    temp_raw.load_data()
+    picks = pick_types(temp_raw.info, meg=True, eeg=True,
+                       stim=True, ecog=True, seeg=True,
+                       exclude='bads')
+    ch_names = [temp_raw.ch_names[k] for k in picks]
+    temp_raw = temp_raw.pick_channels(ch_names)
+
+    # first bandpass the signal to just between 45 and 65 Hz
+    sfreq = temp_raw.info['sfreq']
+    l_freq = 45
```

Comment:
assumes that no data with samples would come in with data < 90 Hz sampling rate. I can speak for EEG/iEEG and say that is never going to happen, but not sure about other modalities?

Reply:
you can add a check for sampling rate, no?